forum/src/main/java/forum/services/VoteService.java

```diff
@@ -0,0 +1,66 @@
+package forum.services;
+
+import forum.entities.Vote;
+import forum.repositories.VoteRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class VoteService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(VoteService.class);
+
+    private final VoteRepository voteRepository;
+
+    @Autowired
+    public VoteService(VoteRepository voteRepository) {
+        this.voteRepository = voteRepository;
+    }
+
+    @Transactional
```

Comment:
Transactional is useless here, cause repository will open transaction for DB call